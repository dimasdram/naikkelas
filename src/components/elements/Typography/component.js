import React from 'react';
import PropTypes from 'prop-types';
import { Typography } from '@material-ui/core';
import classNames from 'classnames';
import { capitalizeFirst } from '../../../utils/text';

export default class Component extends React.Component {
  render() {
    const {
      children,
      classes,
      className,
      color,
      variant,
      ...other
    } = this.props;
    const customClass = classNames([
      classes[variant],
      classes[`color${capitalizeFirst(color)}`],
      className,
      classes.font
    ]);

    return (
      <Typography className={customClass} {...other}>
        {children}
      </Typography>
    );
  }
}

Component.defaultProps = {
  children: 'text',
  className: '',
  color: 'black',
  variant: 'h1'
};

Component.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.number,
    PropTypes.string
  ]),
  classes: PropTypes.object.isRequired,
  className: PropTypes.string,
  color: PropTypes.oneOf([
    'black',
    'blackGrey',
    'white',
    'grey',
    'error',
    'blue',
    'orange',
    'green'
  ]),
  variant: PropTypes.oneOf([
    'button',
    'buttonSmall',
    'caption',
    'captionParagraph',
    'contentLabelParagraph',
    'captionParagraphFooter',
    'contentParagraph1',
    'contentParagraph2',
    'footer',
    'formCaptionParagraph',
    'formLabel',
    'formLabelCap',
    'formLabelSecondary',
    'formLabelTooltip',
    'h1',
    'h2',
    'headFormLabel',
    'labelBold',
    'subLevel',
    'subLevelBold',
    'text',
    'textHead',
    'ubuntuFontStyle'
  ])
};
