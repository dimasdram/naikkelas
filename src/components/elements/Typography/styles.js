const styles = {
  button: {
    fontSize: '16px',
    fontWeight: 'bold',
    textTransform: 'uppercase'
  },
  buttonSmall: {
    fontSize: '14px',
    fontWeight: 'bold'
  },
  caption: {
    fontSize: '14px',
    fontWeight: 'normal'
  },
  captionParagraph: {
    fontSize: '11px',
    fontWeight: '300'
  },
  captionParagraphFooter: {
    color: '#33332F !important',
    fontSize: '11px',
    fontWeight: '300'
  },
  colorBlack: {
    color: '#000 !important'
  },
  colorblackGrey: {
    color: '#33332F !important'
  },
  colorBlue: {
    color: '#2F80ED !important'
  },
  colorError: {
    color: '#ED2A23 !important'
  },
  colorGreen: {
    color: '#369A53 !important'
  },
  colorGrey: {
    color: ' #656563 !important'
  },
  colorOrange: {
    color: '#F89B2B !important'
  },
  colorWhite: {
    color: '#fff !important'
  },
  contentLabelParagraph: {
    fontSize: '14px',
    fontWeight: 'bold',
    letterSpacing: '0.1em',
    lineHeight: '16px'
  },
  contentParagraph1: {
    fontSize: '14px',
    fontWeight: 'normal',
    lineHeight: '300%'
  },
  contentParagraph2: {
    fontSize: '14px',
    fontWeight: 'normal',
    letterSpacing: '0.1em',
    lineHeight: '150%'
  },
  error404: {
    fontSize: '24px',
    fontWeight: 'normal',
    letterSpacing: '0.1em',
    lineHeight: '28px',
    textTransform: 'uppercase'
  },
  font: {
    fontFamily: 'Ubuntu',
    fontStyle: 'normal'
  },
  footer: {
    fontSize: '10px',
    letterSpacing: '0.1em',
    lineHeight: '12px'
  },
  formCaptionParagraph: {
    fontSize: '11px',
    fontWeight: '300'
  },
  formLabel: {
    fontSize: '14px',
    fontWeight: 'bold',
    letterSpacing: '0.1em',
    lineHeight: '16px'
  },
  formLabelCap: {
    fontSize: '12px',
    fontWeight: 'normal',
    letterSpacing: '0.1em',
    lineHeight: '14px'
  },
  formLabelSecondary: {
    fontSize: '14px',
    fontWeight: 'normal',
    letterSpacing: '0.1em',
    lineHeight: '16px'
  },
  formLabelTooltip: {
    fontSize: '11px',
    fontWeight: 'normal'
  },
  h1: {
    fontSize: '24px',
    fontWeight: 'bold',
    margin: '70px',
    textAlign: 'center',
    textTransform: 'uppercase'
  },
  h2: {
    fontSize: '16px',
    fontWeight: 'bold',
    letterSpacing: '0.1em',
    lineHeight: '100%',
    margin: '70px',
    textAlign: 'center',
    textTransform: 'uppercase'
  },
  headFormLabel: {
    fontSize: '18px',
    fontWeight: 'bold',
    letterSpacing: '0.1em',
    lineHeight: '21px'
  },
  labelBold: {
    fontSize: '12px',
    fontWeight: 'bold',
    letterSpacing: '0.1em',
    lineHeight: '14px'
  },
  pageNotfound: {
    fontSize: '36px',
    fontWeight: 'bold',
    letterSpacing: '0.1em',
    lineHeight: '41px'
  },
  subLevel: {
    fontSize: '13px',
    fontWeight: 'normal',
    letterSpacing: '0.1em',
    lineHeight: '15px'
  },
  subLevelBold: {
    fontSize: '13px',
    fontWeight: 'bold',
    letterSpacing: '0.1em',
    lineHeight: '14px'
  },
  text: {
    fontSize: '18px',
    lineHeight: '22px'
  },
  textHead: {
    fontSize: '24px',
    fontWeight: 'bold'
  },
  ubuntuFontStyle: {
    alignItems: 'center',
    display: 'flex',
    fontSize: '18px',
    letterSpacing: '0.1em',
    lineHeight: '22px'
  }
};

export default styles;
