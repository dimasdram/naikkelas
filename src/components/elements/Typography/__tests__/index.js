import React from 'react';
import renderer from 'react-test-renderer';
import Typography from '../index';
import { shallow } from 'enzyme';
jest.mock('react-router-dom');

describe('Typography', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Typography>test</Typography>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('check children', () => {
    const children = 'test';
    const tree = shallow(<Typography>{children}</Typography>);
    expect(tree.props().children === children).toBe(true);
  });

  ['black', 'white', 'grey', 'error', 'blue'].forEach(color => {
    it(`render with color ${color}`, () => {
      const container = 'test';
      const tree = shallow(<Typography color={color}>{container}</Typography>);
      expect(tree.exists()).toBe(true);
    });
  });

  [
    'button',
    'buttonSmall',
    'caption',
    'captionParagraph',
    'contentLabelParagraph',
    'contentParagraph1',
    'contentParagraph2',
    'footer',
    'formCaptionParagraph',
    'formLabel',
    'formLabelCap',
    'formLabelSecondary',
    'formLabelTooltip',
    'h1',
    'h2',
    'headFormLabel',
    'subLevel',
    'subLevelBold',
    'text',
    'textHead',
    'ubuntuFontStyle'
  ].forEach(variant => {
    it(`render with variant ${variant}`, () => {
      const container = 'test';
      const tree = shallow(
        <Typography variant={variant}>{container}</Typography>
      );
      expect(tree.exists()).toBe(true);
    });
  });
});
