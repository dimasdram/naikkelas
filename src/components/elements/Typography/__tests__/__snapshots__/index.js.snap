// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography renders correctly 1`] = `
<p
  className="MuiTypography-root-35 MuiTypography-body1-44 Component-h1-25 Component-colorBlack-6 Component-font-18"
>
  test
</p>
`;
